pkg/metrics/statsd/cosmosdb/metrics.go

```diff
@@ -34,6 +34,9 @@ func (t *tracerRoundTripper) RoundTrip(req *http.Request) (resp *http.Response,
 	start := time.Now()
 
 	defer func() {
+		if resp == nil {
```

Comment:
you are not ever setting resp! ;-)

Reply:
`func (t *tracerRoundTripper) RoundTrip(req *http.Request) (resp *http.Response, err error) ` naked return ?

Reply:
Good point!  hehe.  This makes me think that we should add another metric that counts errors, wdyt?

Reply:
Yes, this was on my head too but I wanted to remove panics first

Reply:
https://github.com/Azure/ARO-RP/issues/131